worlds/undertale/Rules.py

```diff
@@ -0,0 +1,269 @@
+from ..generic.Rules import set_rule, add_rule
+from BaseClasses import MultiWorld
+from ..AutoWorld import LogicMixin
+import typing
+import math
+
+
+class UndertaleLogic(LogicMixin):
```

Comment:
Avoid using `LogicMixin`, instead make distinct functions for logic and call those when setting rules. See Rogue Legacy or Subnautica world definitions for examples. `LogicMixin`s pollute the MultiWorld object with each rule and can cause generation issues if 2 worlds use the same name. Probably not a problem here, since all the functions are prefixed with "_undertale", but still not best practice. It's really there for worlds that have no choice, like Ocarina of Time and Super Metroid.